samza-core/src/main/java/org/apache/samza/container/grouper/stream/AllSspToSingleTaskGrouperFactory.java

```diff
@@ -19,51 +19,116 @@
+ * consumed by each TaskInstance. This is useful, in case of using load-balanced consumers like the high-level Kafka
+ * consumer and Kinesis consumer, where Samza doesn't control the partitions being consumed by the task.
  *
- * Note: This grouper does not take in broadcast streams yet.
+ * Note:
+ * 1. This grouper does not take in broadcast streams yet.
+ * 2. This grouper is supported only with Yarn, Zk-standalone and Passthrough JobCoordinators.
  */
+
 class AllSspToSingleTaskGrouper implements SystemStreamPartitionGrouper {
+  private final boolean isPassthroughJobCoordinator;
+  private final int taskCount;
   private final int containerId;
 
-  public AllSspToSingleTaskGrouper(int containerId) {
+  AllSspToSingleTaskGrouper(boolean isPassthroughJobCoordinator, int containerId, int taskCount) {
+    this.isPassthroughJobCoordinator = isPassthroughJobCoordinator;
     this.containerId = containerId;
+    this.taskCount = taskCount;
   }
 
   @Override
   public Map<TaskName, Set<SystemStreamPartition>> group(final Set<SystemStreamPartition> ssps) {
+    Map<TaskName, Set<SystemStreamPartition>> groupedMap = new HashMap<>();
+
     if (ssps == null) {
       throw new SamzaException("ssp set cannot be null!");
     }
     if (ssps.size() == 0) {
       throw new SamzaException("Cannot process stream task with no input system stream partitions");
     }
 
-    final TaskName taskName = new TaskName(String.format("Task-%s", String.valueOf(containerId)));
+    if (isPassthroughJobCoordinator) {
+      final TaskName taskName = new TaskName(String.format("Task-%d", containerId));
+      groupedMap.put(taskName, ssps);
+    } else {
+      // Assign all partitions to each task
+      IntStream.range(0, taskCount).forEach(i -> {
+          final TaskName taskName = new TaskName(String.format("Task-%s", String.valueOf(i)));
+          groupedMap.put(taskName, ssps);
+        });
+    }
 
-    return Collections.singletonMap(taskName, ssps);
+    return groupedMap;
   }
 }
 
 public class AllSspToSingleTaskGrouperFactory implements SystemStreamPartitionGrouperFactory {
+  private static final String YARN_CONTAINER_COUNT = "yarn.container.count";
+  private static final String PASSTHROUGH_JOB_COORDINATOR_FACTORY = PassthroughJobCoordinatorFactory.class.getName();
+  private static final String ZK_JOB_COORDINATOR_FACTORY = ZkJobCoordinatorFactory.class.getName();
+
   @Override
   public SystemStreamPartitionGrouper getSystemStreamPartitionGrouper(Config config) {
-    return new AllSspToSingleTaskGrouper(config.getInt(JobConfig.PROCESSOR_ID()));
+    int taskCount;
+    int containerId = 0;
+    boolean isPassthroughJobCoordinator = false;
+    String jobCoordinatorFactoryClassName = config.get(JobCoordinatorConfig.JOB_COORDINATOR_FACTORY);
+
+    if (jobCoordinatorFactoryClassName != null && !jobCoordinatorFactoryClassName.isEmpty()) {
+      if (jobCoordinatorFactoryClassName.equals(ZK_JOB_COORDINATOR_FACTORY)) {
```

Comment:
A further modification to make the code simpler would be to add an interface method getProcessorNames() to JobCoordinatorFactory class and let each implementation class to implement it (or throw exception by default). That would avoid all the if-then-else cases here and avoid copying ZkJobCoordinatorFactory code here.

Reply:
+1! Added the API just like you suggested.